tstest/integration/testcontrol/testcontrol.go

```diff
@@ -101,6 +102,19 @@ func (s *Server) serveUnhandled(w http.ResponseWriter, r *http.Request) {
 	go panic(fmt.Sprintf("testcontrol.Server received unhandled request: %s", got.Bytes()))
 }
 
+func (s *Server) addPingRequest(res *tailcfg.MapResponse) error {
+	if len(res.Peers) == 0 {
+		return errors.New("MapResponse has no peers to ping")
+	}
+
+	if len(res.Peers[0].Addresses) == 0 || len(res.Peers[0].AllowedIPs) == 0 {
+		return errors.New("peer has no Addresses or no AllowedIPs")
+	}
+	targetIP := res.Peers[0].AllowedIPs[0].IP()
```

Comment:
I'd think you'd want a bit of control over this, instead of using whatever happens to be the first allowed IP of whatever happens to be the first peer. If there's always going to be exactly one of each, then I'd suggest enforcing that here. But it strikes me as preferable to pass the peer+allowed IP into the endpoint and thread them through here, as that's much more like what the real control server will be doing.

Reply:
It now sends the Node* as part of the Request